examples/ex9.cpp

```diff
@@ -60,22 +60,277 @@ Vector bb_min, bb_max;
+   DenseMatrix velEval;
+   
+   elDiff.SetSize(ne); elDiff = 0.;
+   lumpedM.SetSize(ne); lumpedM = 0.;
+   adjJ.SetSize(dim,dim);
+   
+   for (k = 0; k < ne; k++)
+   {
+      ///////////////////////////
+      // Element contributions //
+      ///////////////////////////
+      const FiniteElement &el = *fes->GetFE(k);
+      nd = el.GetDof();
+      tr = mesh->GetElementTransformation(k);
+      // estim1 can not be integrated exactly due to transforamtion dependent denominator
+      // use tr->Order()-1 + 4*el.GetOrder() instead
+      // appropriate qOrd for estim2 is tr->Order()-1 + 2*el.GetOrder(), choose max
+      qOrd = tr->Order()-1 + 4*el.GetOrder();
+      const IntegrationRule *ir = &IntRules.Get(el.GetGeomType(), qOrd);
+      
+      shape.SetSize(nd);
+      dshape.SetSize(nd,dim);
+      estim1.SetSize(nd);
+      estim2.SetSize(nd);
+      vec1.SetSize(nd);
+      vec2.SetSize(nd);
+      estim1 = estim2 = 0.;
+      
+      coef.Eval(velEval, *tr, *ir);
+            
+      for (p = 0; p < ir->GetNPoints(); p++)
+      {
+         const IntegrationPoint &ip = ir->IntPoint(p);
+         tr->SetIntPoint(&ip);
+
+         el.CalcDShape(ip, dshape);
+         CalcAdjugate(tr->Jacobian(), adjJ);
+         el.CalcShape(ip, shape);
+         
+         velEval.GetColumnReference(p, vval);
+         adjJ.Mult(vval, vec1);
+         dshape.Mult(vec1, vec2);
+         for (j = 0; j < nd; j++)
+         {
+            //divide due to square in L2-norm
+            estim1(j) += ip.weight / tr->Weight() * pow(vec2(j), 2.); 
+            estim2(j) += ip.weight * tr->Weight() * pow(shape(j), 2.);
+         }
+         lumpedM(k) += ip.weight * tr->Weight();
+      }
+      elDiff(k) = std::sqrt(estim1.Max() * estim2.Max());
+      lumpedM(k) /= nd;
+      
+      ////////////////////////
+      // Face contributions //
+      ////////////////////////
+      if (dim==1)
+         mesh->GetElementVertices(k, bdrs);
+      else if (dim==2)
+         mesh->GetElementEdges(k, bdrs, orientation);
+      else if (dim==3)
+         mesh->GetElementFaces(k, bdrs, orientation);
+      
+      numBdrs = bdrs.Size();
+      geomType = el.GetGeomType();
+
+      if (k==0)
+         bdrDiff.SetSize(ne,numBdrs);
+
+      for (i = 0; i < numBdrs; i++)
+      {
+         extractBdrDofs(geomType, nd, i, dofs);
+         numDofs = dofs.Size();
+         
+         FaceElementTransformations *Trans = mesh -> GetFaceElementTransformations(bdrs[i]); 
+         // qOrd is chosen such that L2-norm of basis is computed accurately.
+         // Normal velocity term relies on L^{infty}-norm which is approximated 
+         // by its maximum value in the quadrature points of the same rule.
+         if (Trans->Elem1No != k)
+         {
+            if (Trans->Elem2No != k)
+               mfem_error("Boundary edge does not belong to this element.");
+            else
+               qOrd = Trans->Loc2.Transf.Order() + 2*el.GetOrder();
```

Comment:
Does `Trans->Loc2.Transf.Order()` is to take in account the order of the determinant of the Jacobian? I'm not sure but the terms in the Jacobian are of order: `Order()-1` at most in each column for each variable? Which results in a determinant of order: `(Order()-1)*Dim` at most for the determinant. But I might be wrong.
I've build this artificial example: `f(x,y)->(x^4,yx^3)` then `det(J_f)=4x^6`, then `order(f)=4` but `order(det(J_f))=6`.
Don't change your code because of this, I have no idea how relevant what I'm saying is.

Reply:
Actually it should even be (order()-1)^dim, I didn't change it as you said but I guess we should talk about quadrature orders in general. For remap I suppose you have a variable that describes the order of the grid velocity that should also go into the order?

Reply:
To me the determinant is a product of `Dim` polynomials of order `Order() - 1` at most, which results in a polynomial of order `(Order() - 1)*Dim` or am I missing something?
Anyway, I haven't much experience with the orders of quadrature rules to use on curved meshes, I believe that usually terms are under-integrated in MFEM, but maybe @v-dobrev has some information about this.

Reply:
My bad, of course you're right about (Order() - 1)*Dim.

Reply:
`Loc2` is the local face to volume transformation (which is always linear) - you probably want to use something like
https://github.com/mfem/mfem/blob/eea8eb2185c5b6368ca2c1c15414344549e67b63/fem/bilininteg.cpp#L2289-L2295
Also, note that the transformation class has a method to compute the order of the determinant - `OrderW()`.